packages/experimental/RadioGroup/src/RadioGroup/useRadioGroup.ts

```diff
@@ -20,13 +20,15 @@ export const useRadioGroup = (props: RadioGroupProps): RadioGroupInfo => {
 
   const state: RadioGroupState = {
     value: data.selectedKey,
+    required: required || false,
     onChange: data.onKeySelect,
     updateSelectedButtonRef: onSelectButtonRef,
   };
 
   return {
     props: {
       ...props,
+      required,
       accessible: true,
       accessibilityRole: 'radiogroup',
       accessibilityLabel: accessibilityLabel ?? label,
```

Comment:
I think there needs to be a required accessibilityState set for screen readers to correctly announce the required state of a radio group. You can look at Checkbox as an example.

Reply:
Added!